modules/core/src/shaderlib/project/project-functions.js

```diff
@@ -37,25 +81,27 @@ export function projectPosition(
     modelMatrix,
     fromCoordinateSystem,
     fromCoordinateOrigin
-  }
-) {
-  let [x, y, z] = position;
-  let worldPosition;
+  } = normalizeParameters(params);
 
   if (modelMatrix) {
     [x, y, z] = vec4_transformMat4([], [x, y, z, 1.0], modelMatrix);
   }
-  if (fromCoordinateSystem === undefined) {
-    fromCoordinateSystem = coordinateSystem;
-  }
-  if (fromCoordinateOrigin === undefined) {
-    fromCoordinateOrigin = coordinateOrigin;
+
+  let offsetMode = false;
+  switch (coordinateSystem) {
```

Comment:
Nit: Add all enums here and assert on default, it will make things easier when new enums area added.